vllm/v1/structured_output/utils.py

```diff
@@ -90,6 +90,9 @@ def apply_grammar_bitmask(
     seq = sorted(scheduler_output.structured_output_request_ids.items(),
                  key=lambda x: x[1])
     for req_id, _ in seq:
+        if req_id not in struct_out_req_batch_indices:
+            # preempted req ids are not in input_batch, we skip in this case
+            continue
```

Comment:
seems like for a preempted request, the request ID should not be included in the scheduler output as an active structured output request. Maybe we should fix this on the scheduler side?